pkg/operator/controller/nodeobservability/daemonset.go

```diff
@@ -32,9 +35,16 @@ func (r *NodeObservabilityReconciler) ensureDaemonSet(ctx context.Context, nodeO
 	desired := r.desiredDaemonSet(nodeObs, sa)
 	exist, current, err := r.currentDaemonSet(ctx, nameSpace)
 	if err != nil {
-		return false, nil, fmt.Errorf("failed to get DaemonSet: %v", err)
+		return false, nil, fmt.Errorf("failed to get DaemonSet: %w", err)
 	}
 	if !exist {
+		cmExists, err := r.createConfigMap(nodeObs)
```

Comment:
I understood the the first boolean is to indicate the creation of the configmap, not the existence.